### PR TITLE
tests: Checks for Windows permissions as well

### PR DESCRIPTION
**Pull request by claudiubelu**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
/kind failing-test

**What this PR does / why we need it**:

Some mounttest related tests are checking the file permissions set on the
container files, but the default file permissions on Windows is 775 instead of
644, causing some tests to fail.

Keep in mind that file permissions work differently on Windows, and setting file
permissions via Kubernetes is not currently supported on Windows.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related issue: #70879

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jsafrane** [comment]
+ check `hack/verify-gofmt.sh` output, it's not happy about your coding style :-)

File: test/e2e/common/configmap_volume.go [hunk 1]
```diff
@@ -622,17 +621,14 @@ func doConfigMapE2EWithoutMappings(f *framework.Framework, uid, fsGroup int64, d
...
-		mode := int32(0644)
-		defaultMode = &mode
 	}
 
-	modeString := fmt.Sprintf("%v", os.FileMode(*defaultMode))
+	fileModeRegexp := framework.GetFileModeRegex("/etc/configmap-volume/data-1", defaultMode)
 	output := []string{
-		"content of file \"/etc/configmap-volume/data-1\": value-1",
-		"mode of file \"/etc/configmap-volume/data-1\": " + modeString,
+		"content of file \"\\/etc\\/configmap-volume\\/data-1\": value-1",
```
Comment (ixdy) [inline test/e2e/common/configmap_volume.go]: nit: you can avoid some of the escaping here by using raw strings, e.g.
```golang
`content of file "\/etc\/configmap-volume\/data-1": value-1`
```

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -5252,3 +5252,37 @@ func WaitForNodeHasTaintOrNot(c clientset.Interface, nodeName string, taint *v1.
...
+	)
+	if mask == nil {
+		linuxMask = int32(0644)
+		windowsMask = int32(0775)
+	} else {
+		linuxMask = *mask
+		windowsMask = *mask
+	}
+
+	// forward slashes have to be escaped for regexp
```
Comment (ixdy) [inline test/e2e/framework/util.go]: is this something special imposed by gomega? AFAIK the golang regexp library doesn't treat `/` special.

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -5252,3 +5252,37 @@ func WaitForNodeHasTaintOrNot(c clientset.Interface, nodeName string, taint *v1.
...
+}
+
+func getFileAttributeRegex(attributeName string, filePath string, mask *int32) string {
+	var (
+		linuxMask   int32
+		windowsMask int32
+	)
+	if mask == nil {
+		linuxMask = int32(0644)
+		windowsMask = int32(0775)
```
Comment (ixdy) [inline test/e2e/framework/util.go]: the comment says 0755, but this is 0775 - which is incorrect?

File: test/e2e/common/configmap_volume.go (hunk 1 above)
Comment (ixdy) [reply test/e2e/common/configmap_volume.go]: also, in `getFileAttributeRegex`, it reads `mode of Windows file`. should there be a `Windows` here too?

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -5252,3 +5252,37 @@ func WaitForNodeHasTaintOrNot(c clientset.Interface, nodeName string, taint *v1.
...
+
+// GetFileModeRegex returns a file mode related regex which should be matched by the mounttest pods' output.
+// If the given mask is nil, then the regex will contain the default OS file modes, which are 0644 for Linux and 0755 for Windows.
+func GetFileModeRegex(filePath string, mask *int32) string {
+	return getFileAttributeRegex("mode", filePath, mask)
+}
+
+// GetFilePermsRegex returns a file permissions related regex which should be matched by the mounttest pods' output.
+// If the given mask is nil, then the regex will contain the default OS file permissions, which are 0644 for Linux and 0755 for Windows.
+func GetFilePermsRegex(filePath string, mask *int32) string {
```
Comment (ixdy) [inline test/e2e/framework/util.go]: it doesn't look like anything uses this function - do we need it?

File: test/e2e/common/configmap_volume.go (hunk 1 above)
Comment (bclau) [reply test/e2e/common/configmap_volume.go]: Done.

The file content is still the same, so there is no need to change the expected output related to the file content. This patch only addresses the differences of default file permissions between Linux and Windows.

File: test/e2e/framework/util.go (hunk 3 above)
Comment (bclau) [reply test/e2e/framework/util.go]: 0775, my bad. :)

File: test/e2e/framework/util.go (hunk 2 above)
Comment (bclau) [reply test/e2e/framework/util.go]: Hm, there might have been a reason why this was done in the 1st place, but I no longer remember. Seems to work without it, so I'll just remove it.

File: test/e2e/framework/util.go [hunk 5]
```diff
@@ -5252,3 +5252,24 @@ func WaitForNodeHasTaintOrNot(c clientset.Interface, nodeName string, taint *v1.
 	}
 	return nil
 }
+
+// GetFileModeRegex returns a file mode related regex which should be matched by the mounttest pods' output.
+// If the given mask is nil, then the regex will contain the default OS file modes, which are 0644 for Linux and 0755 for Windows.
```
Comment (ixdy) [inline test/e2e/framework/util.go]: update the comment to `0775` here?

File: test/e2e/common/configmap_volume.go [hunk 6]
```diff
@@ -622,17 +621,14 @@ func doConfigMapE2EWithoutMappings(f *framework.Framework, uid, fsGroup int64, d
...
-		mode := int32(0644)
-		defaultMode = &mode
 	}
 
-	modeString := fmt.Sprintf("%v", os.FileMode(*defaultMode))
+	fileModeRegexp := framework.GetFileModeRegex("/etc/configmap-volume/data-1", defaultMode)
 	output := []string{
-		"content of file \"/etc/configmap-volume/data-1\": value-1",
-		"mode of file \"/etc/configmap-volume/data-1\": " + modeString,
+		`content of file "\/etc\/configmap-volume\/data-1": value-1`,
```
Comment (ixdy) [inline test/e2e/common/configmap_volume.go]: maybe remove the escapes from the `/`s here (and in similar lines throughout), if they aren't needed?

Comment (bclau) [reply test/e2e/common/configmap_volume.go]: Done

**Pull request merged**